src/lib/modules/blockchain_connector/provider.js

```diff
@@ -16,9 +20,31 @@ class Provider {
 
   startWeb3Provider(callback) {
     const self = this;
+    const accountsConfig = self.accountsConfig[0];
 
     if (this.type === 'rpc') {
       self.provider = new this.web3.providers.HttpProvider(self.web3Endpoint);
+    } else if (this.type === 'ledger') {
+        if (accountsConfig.HardwareWallet === 'Ledger') {
+
+          const rpcUrl = accountsConfig.HardwareWallet.RpcUrl;
+          const engine = new ProviderEngine();
+          const getTransport = () => TransportNodeHid.open(accountsConfig.LedgerWalletHardwareAddress);
+          const ledger = createLedgerSubprovider(getTransport, {
+            networkId: accountsConfig.NetworkId,  // refer to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+            accountsLength: accountsConfig.NoOfLedgerAccountsToLoad // number of accounts to load from the device
+          });
+
+          engine.addProvider(ledger); 
+          engine.addProvider(new FetchSubprovider({ rpcUrl }));
+          engine.start();
+          self.provider = engine;
+
+          // listen for hardware connectivity error
+          self.provider.on('error', (err) => self.logger.error('Error while connecting to the ledger wallet ', (err.message || err.stack) || err));
+          self.provider.on('end', () => self.logger.error('Connection to wallet ended'));
+
+        }
```

Comment:
I cannot quite comment on it because it is too low, but below, we call `AccountParser.parseAccountsConfig`, which checks the `accountsConfig`. Since you added a new account type (ledger) and it's not supported by the AccountParser (since it's handled by the provider), it will give a warning saying `Unsupported account configuration`. You should add a condition in `AccountParser.getAccountgetAccount()` that just ignores the `ledger` account.

Reply:
@jrainville good catch on this 🙂 i had actually disabled the AccountParser part when testing on my end because i was getting the `Unsupported account configuration: ` msg. I forgot i needed to do this check when i re-enabled it in order to push.